fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_IdentPage.js

```diff
@@ -62,11 +64,49 @@ function calculateSecDate(date) {
     } else {
         return "ERR";
     }
+}
+
+function switchDataBase(mcdu)
+    {
+    // Only performing a reset of the MCDU for now, no secondary database
+    //Speed AP returns to selected
+    const isSelected = Simplane.getAutoPilotAirspeedSelected();
+    if (isSelected == false) SimVar.SetSimVarValue("H:A320_Neo_FCU_SPEED_PULL", "boolean", 1);
+    //flight plan
+    mcdu.resetCoroute();
+    mcdu.atsu.atc.resetAtisAutoUpdate();
+    mcdu.flightPlanManager.clearFlightPlan();
```

Comment:
This function is async. You need to await it, which might explain your other problems with flightplan waypoints going away while they're being iterated in deleteAllStoredWaypoints.

Reply:
Resolved by new code. using .then() to avoid async and await on my function